keps/sig-api-machinery/2211-strip-managed-fields-server-side/README.md

```diff
@@ -0,0 +1,426 @@
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+ManagedFields is used to keep track of who has changed a field of an object, and Server-side Apply relies on such information.
+While the managedFields is beneficial to resolving conflicts on updates, it is a bit verbose to many users because this field
+could be quite large and generally users are not interested in this information.
+
+This KEP is proposing stripping the managedFields on the server-side when users request objects to make the response less verbose.
+
+## Motivation
+
+Many users have complained in [kubernetes#90066][90066] that they got impacted by the addition of managedFields
+(and the issue itself has received more than 100 thumb-up reactions), and we want to improve the user experience.
+
+[90066]: https://github.com/kubernetes/kubernetes/issues/90066
+
+At the same time, it is not ideal to only drop the managedFields on the client-side, because that would require every client
+(such as kubectl or CLI tools written by users using client-go) to do the same thing.
+
+With this KEP, the API server could return objects without managedFields on demand, so that least client-side changes is needed
+to make the response less verbose. 
+
+### Goals
+
+* Strip managedFields on the server-side.
+
+### Non-Goals
+
+* Turn off managedFields. Server-side Apply relies on managedFields, we could not simply disable it.
+* Clear manageFields only on the client-side.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### Server-side changes
+
+* Add three content types to denote the client expects the API server to strip the managedFields.
+  * `application/json+unmanaged`
+  * `application/yaml+unmanaged`
+  * `application/vnd.kubernetes.protobuf+unmanaged`
```

Comment:
Even if we don't introduce it just yet, we might have other versions for the encoding of managed fields in the future. Currently, we only have `v1`. Can we describe how the header would look like if I wanted to request a specific format version of managed fields in the future?

Reply:
"unmanaged" isn't the right word. "omitting.managed.fields", "sans-managed" perhaps.

Reply:
I'd prefer something that was a positive specification rather than negative. So, like we could add "managed.v1" (the default) and "managed.omit" (the new option), which leaves room for v2 etc.

Reply:
This needs to compose with the tabular output content type.

Also I think clients would specify these in the "Accept" header. Ah, it's mentioned below. But better to be specific.

Reply:
> Can we describe how the header would look like if I wanted to request a specific format version of managed fields in the future

I guess this is out of the scope of this KEP, but we could still leave room for it. 

@lavalamp IIUC are you suggesting changing the `Content-Type` to `application/{json,yaml,profobuf}.managed.omit`?

Reply:
let's also look at this really quick: https://trac.tools.ietf.org/html/draft-ietf-appsawg-media-type-regs-14#section-6